src/components/FavoriteFilter/FavoriteFilter.test.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import FavoriteFilter from './FavoriteFilter'
+
+describe('FavoriteFilter', () => {
+  let wrapper
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <FavoriteFilter number={2} />)
+  })
+
+  it('should render FavoritesFilter with all data passed in correctly and toFavorite state is false', () => {
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  it('should render FavoritesFilter with Redirect component if toFavorite state is true', () => {
+    wrapper.setState({toFavorite: true})
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  it('should call redirect with all data passed in correctly', () => {
+    const spy = jest.spyOn(wrapper.instance(), 'redirect').mockImplementation(arg => arg)
```

Comment:
Ok to check my understanding here - when you use a spyOn method, it takes the thing you're spying on and the method you want to look at as arguments.  Then with mockImplementation you're basically telling the spy to run the second argument (the redirect method) as is.

Then you force the instance to update, simulate a click and check to make sure redirect was called?

Reply:
first i make spy on instance method as a mock fn, update instance for change original fn to mock equivalent, and check that this fn has been called